app/uk/gov/hmrc/gform/testonly/TestOnlyController.scala

```diff
@@ -519,6 +527,34 @@ class TestOnlyController(
       )
   }
 
+  def restoreAll(formTemplateId: FormTemplateId, maybeAccessCode: Option[AccessCode]) =
+    auth.authWithoutRetrievingForm(formTemplateId, OperationWithoutForm.ShowAccessCode) {
+      implicit request => implicit lang => cache =>
+        val snapshotId = request.body.asFormUrlEncoded.get("snapshotId").head
+        val formTemplateContext = request.attrs(FormTemplateKey)
+        for {
+          // create a brand new form with snapshot's template id
+          _                <- gformConnector.restoreSnapshotTemplate(snapshotId)
+          _                <- newFormController.continue(cache, formTemplateContext.formTemplate)
+          snapshotWithData <- gformConnector.snapshotData(snapshotId)
```

Comment:
I think it retrieves only `formTemplateId` but we have it in this func. Is it required?

Reply:
yes, you are right. I removed it.